app/assets/javascripts/notifications/listener-events.coffee

```diff
@@ -0,0 +1,331 @@
+      'id',          # Number
+      'name',        # String
+      'code',        # String
+      'isForever',   # Boolean
+      'isNowRunning' # Boolean, false if not a forever button or a forever button was turned off
+    ]
+  },
+  {
+    'name': 'slider-widget-changed',
+    'args': [
+      'id',       # Number
+      'global',   # String
+      'newValue', # Number, the new value of the widget
+      'oldValue'  # Number, the prior value of the widget
+    ]
+  },
+  {
+    'name': 'chooser-widget-changed',
+    'args': [
+      'id',       # Number
+      'global',   # String
+      'newValue', # Any, the new value of the widget
+      'oldValue', # Any, the prior value of the widget
+    ]
+  },
+  {
+    'name': 'input-widget-changed',
+    'args': [
+      'id',       # Number
+      'global',   # String
+      'newValue', # Any, the new value of the widget
+      'oldValue', # Any, the prior value of the widget
+      'type'      # 'Number' | 'String' | 'String (reporter)' | 'String (command)' | 'Color'
+    ]
+  },
+  {
+    'name': 'switch-widget-changed',
+    'args': [
+      'id'        # Number
+      'global',   # String
+      'newValue', # Boolean, the new value of the widget
+      'oldValue', # Boolean, the prior value of the widget
+    ]
+  },
+  {
+    'name': 'command-center-run',
+    'args': [
+      'command' # String
+    ]
+  },
+  {
+    'name': 'speed-slider-changed',
+    'args': [
+      'speed' # Number, the new speed from -1 (slowest) to 1 (fastest)
+    ]
+  },
+  {
+    'name': 'command-center-toggled',
+    'args': [
+      'isOpen' # Boolean
+    ]
+  },
+  {
+    'name': 'model-code-toggled',
+    'args': [
+      'isOpen' # Boolean
+    ]
+  },
+  {
+    'name': 'model-info-toggled',
+    'args': [
+      'isOpen' # Boolean
+    ]
+  },
+  {
+    'name': 'title-changed',
+    'args': [
+      'title' # String
+    ]
+  },
+  {
+    'name': 'authoring-mode-toggled',
+    'args': [
+      'isActive' # Boolean
+    ]
+  },
+  {
+    'name': 'notify-user',
+    'args': [
+      'message' # String
+    ]
+  },
+  {
+    'name': 'compiler-error',
+    'args': [
+      'source', # 'recompile' | 'recompile-procedures' | 'export-nlogo' | 'export-html' | 'button' | 'chooser'
+                # | 'slider' | 'plot' | 'input' | 'switch' | 'console'
+      'errors'  # Array[Exception]
+    ]
+  },
+  {
+    'name': 'runtime-error',
+    'args': [
+      'source',   # 'button' | ' console'
+      'exception' # Exception
+    ]
+  },
+  {
+    'name': 'extension-error',
+    'args': [
+      'messages' # Array[String]
+    ]
+  }
+])
+
+getArgName = (argSetting, args) ->
+  if (typeof argSetting) is 'string'
+    argSetting
+  else
+    dependentValue = args[argSetting.sourceArg]
+    maybeCase = argSetting.cases.filter( (argCase) ->
+      argCase.sourceArgValues.includes(dependentValue)
+    )
+    if maybeCase.length is 1
+      foundCase = maybeCase[0]
+      foundCase.argToAdd
+    else
+      null
```

Comment:
The large majority of the upper `else` branch could be replaced with `argSetting.cases.find(f)?.argToAdd`.  This will, however, return `undefined` where it would previously have returned `null`, which will require an edit in `createNamedArgs` below.